### PR TITLE
feat: use path-alias

### PR DESCRIPTION
**Pull request by marcozzxx810**
close: #764

### CODE REVIEW

File: gui/package.json [hunk 1]
```diff
@@ -43,8 +43,7 @@
     "lint": "eslint --max-warnings=0 \"src/**/*.{js,jsx,ts,tsx,json}\" && prettier --check \"src/**/*.{js,jsx,ts,tsx,css,md,json}\"",
     "lint:fix": "eslint --fix \"src/**/*.{js,jsx,ts,tsx,json}\"",
     "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx,css,md,json}\"",
-    "preview-vite": "vite preview",
-    "javaversion-build": "cd src-tauri/src/ && javac JavaVersion.java && jar cvfe JavaVersion.jar JavaVersion JavaVersion.class"
```
Comment (ImUrX) [inline gui/package.json:47]: why did you delete this, its there for a purpose

File: gui/src/components/tracker/TrackerSettings.tsx [hunk 2]
```diff
@@ -10,22 +10,22 @@ import {
...
+import { ArrowLink } from '@/components/commons/ArrowLink';
+import { BodyPartIcon } from '@/components/commons/BodyPartIcon';
+import { Button } from '@/components/commons/Button';
+import { CheckBox } from '@/components/commons/Checkbox';
+import { WarningIcon } from '@/components/commons/icon/WarningIcon';
+import { Input } from '@/components/commons/Input';
+import { Typography } from '@/components/commons/Typography';
+import { MountingSelectionMenu } from '@/components/onboarding/pages/mounting/MountingSelectionMenu';
+import { IMUVisualizerWidget } from '@/components/widgets/IMUVisualizerWidget';
 import { SingleTrackerBodyAssignmentMenu } from './SingleTrackerBodyAssignmentMenu';
```
Comment (ImUrX) [inline gui/src/components/tracker/TrackerSettings.tsx:29]: so if its next to it then we dont do path aliasing? :sweat:

**Review by ImUrX — CHANGES_REQUESTED** [review]

File: gui/src/components/tracker/TrackerSettings.tsx (hunk 2 above)
Comment (ImUrX) [reply gui/src/components/tracker/TrackerSettings.tsx:29]: i dont have a formed opinion on this yet, you can close this one

File: gui/package.json (hunk 1 above)
Comment (TheDevMinerTV) [reply gui/package.json:47]: This ^

File: gui/src/components/tracker/TrackerSettings.tsx (hunk 2 above)
Comment (TheDevMinerTV) [reply gui/src/components/tracker/TrackerSettings.tsx:29]: IMO, we should only use `@` if it actually makes the path shorter.

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]

File: gui/package.json (hunk 1 above)
Comment (marcozzxx810) [reply gui/package.json]: fixed!

File: gui/src/components/tracker/TrackerSettings.tsx (hunk 2 above)
Comment (marcozzxx810) [reply gui/src/components/tracker/TrackerSettings.tsx:29]: the current setting is relative import depth = 1, which means we allow relative import if the file share the same place as the import file

**Review by ImUrX — APPROVED** [review]

**Comment by ImUrX** [comment]
resolve conflicts pls

**Review by Louka3000 — APPROVED** [review]
lgtm

**Review by ButterscotchV — APPROVED** [review]

**Comment by ImUrX** [comment]
more conflicts qwq

**Comment by Louka3000** [comment]
Damn... @marcozzxx810 I would merge but can you take care of conflicts?

firefox_dioBdNkefh

**Comment by marcozzxx810** [comment]
Resolved all the conflict!

**Comment by marcozzxx810** [comment]
@Louka3000 anything that I can help for this pr?

**Comment by Louka3000** [comment]
> @Louka3000 anything that I can help for this pr?

If @ImUrX is fine with the changes, I'm fine with them... I don't really do GUI stuff.
If I were to merge the PR tho, @TheDevMinerTV 's request would need to be resolved (https://github.com/SlimeVR/SlimeVR-Server/pull/771#discussion_r1257344870)

**Review by TheDevMinerTV — CHANGES_REQUESTED** [review]
What I request is still not resolved.

**Comment by marcozzxx810** [comment]
> What I request is still not resolved.

The plugin itself doesnt has this rule, also it is hard to determine shorter when we do static analysis 

the plugin works when there are matching path that has alias

I dont think it make sense to only use when the path is shorter, it is aim to remove ambiguity of relative path.

Think about this

Assume there are two base {Rootdir GUI}/component/button/base

{Rootdir GUI}/component/button/SecondButton/base


In a file {rootdir GUI}/component/button/SecondButton/common/index.ts, you will use '../../base', logically you need to think about which base but '@/component/button/base" directly tell u which base you are referring to

The goal is not shorter, is to clear the ambiguity

**Comment by TheDevMinerTV** [comment]
ah fair

**Review by TheDevMinerTV — APPROVED** [review]
lgtm

**Comment by marcozzxx810** [comment]
merged the latest main!
